### PR TITLE
[TT-5935] Add oas custom marshal function to handle SQL cases

### PR DESCRIPTION
**Pull request by furkansenharputlu**

### CODE REVIEW

File: apidef/oas/oas.go [hunk 1]
```diff
@@ -14,6 +14,25 @@ type OAS struct {
...
+	}
+
+	if s.Info != nil && ShouldOmit(s.Info.License) { // for sql case
+		s.Info.License = nil
+	}
+
+	type Alias OAS
+
+	// to prevent infinite recursion
+	return json.Marshal(&struct {
```
Comment (jeffy-mathew) [inline apidef/oas/oas.go:29]: `OAS.T` already has a `MarshalJSON()` func, shall we use it here?

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4171/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2629114000)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4171/merge
Commit: 7e4626f84593a8b32989b2cb182837353865b612 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2629167505)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4171/merge
Commit: cd9a6909bb3b46b6ac4f96a8bb4499f010554f8b 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2629254043)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/heads/master
Commit: 563a3d459c8fca049e122fdd191ccd992837cc51 [TT-5935] Add oas custom marshal function to handle SQL cases (#4171)

* Add oas custom marshal function to handle SQL cases

* Add small test
Triggered by: push (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2629323157)

**Pull request merged**